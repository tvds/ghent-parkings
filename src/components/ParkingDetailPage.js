
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Map from "./Map";

const ParkingDetailPage = (props) => {
  return (
    <div>
      <div className="detail-header">
        <Link className="detail-header__link" to="/">
          <div className="content-container">
            <FontAwesomeIcon icon={['fas', 'chevron-left']} />  Return to dashboard
          </div>
        </Link>
      </div>
      <div className="detail-body">
        <div className="content-container detail-body__data">
          <h1 className="detail-body__title">
            {props.parking.description}
          </h1>
          <div className="detail-body__sub-title">{props.parking.contactInfo}</div>
          <h3>Opening hours</h3>
          {
            props.parking.openingTimes[0].days.map((day, key) => {
              return <div key={key}>{day}: {props.parking.openingTimes[0].from} - {props.parking.openingTimes[0].to}</div>;
            })
          }
        </div>
        <Map
          center={{ lat: 51.052757, lng: 3.718902 }}
          zoom={13}
          places={[props.parking]}
        />
      </div>
    </div>
  );
}


const mapStateToProp = (state, props) => {
  return {
    parking: state.parkings.find((parking) => parking.id.toString() === props.match.params.id)
  };
};

export default connect(mapStateToProp)(ParkingDetailPage);