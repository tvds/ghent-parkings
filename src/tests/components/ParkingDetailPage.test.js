import React from 'react';
import { shallow } from 'enzyme';
import ParkingDetailPage from '../../components/ParkingDetailPage';

test('should render ParkingDetailPage correctly', () => {
  const wrapper = shallow(<ParkingDetailPage />);
  expect(wrapper).toMatchSnapshot();
});
