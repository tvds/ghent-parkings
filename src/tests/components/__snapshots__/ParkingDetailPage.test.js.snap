// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ParkingDetailPage correctly 1`] = `<[object Object] />`;
